### PR TITLE
mrf24j40: Fix netdev recv implementation

### PR DESCRIPTION
**Pull request by bergzand**
The mrf24j40 driver should return the frame length when both buf is NULL
and len is zero and drop the packet when len is nonzero and buf is NULL.
This PR fixes that behaviour

### Issues/PRs references

None

### CODE REVIEW

File: drivers/mrf24j40/mrf24j40_netdev.c [hunk 1]
```diff
@@ -110,46 +110,45 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
...
-        mrf24j40_rx_fifo_read(dev, phr + 1, &(radio_info->lqi), 1);
-        mrf24j40_rx_fifo_read(dev, phr + 2, &(rssi_scalar), 1);
-        radio_info->rssi = mrf24j40_dbm_from_reg(rssi_scalar);
+        res = pkt_len;
     }
-
     /* Turn on reception of packets off the air */
     mrf24j40_reg_write_short(dev, MRF24J40_REG_BBREG1, 0x00);
-    return pkt_len;
+    /* Return -ENOBUFS if a too small buffer is supplied. Return packet size
```
Comment (PeterKietzmann) [inline drivers/mrf24j40/mrf24j40_netdev.c]: Why don't you need to flush the rx buffer anymore?
`mrf24j40_reg_write_short(dev, MRF24J40_REG_RXFLUSH, MRF24J40_RXFLUSH_RXFLUSH);`

Comment (bergzand) [reply drivers/mrf24j40/mrf24j40_netdev.c]: Page 109 of the mrf24j40 datasheet describes the ways to reset the RX FIFO pointer. writing the RXFLUSH bit is one way, but it is also reset by reading the first byte of the received frame. As we always [read the first bytes of the frame](https://github.com/RIOT-OS/RIOT/pull/9562/files#diff-c64be8fae4f7b67b2428da522851b68eR120) this extra flush is not necessary.

Comment (PeterKietzmann) [reply drivers/mrf24j40/mrf24j40_netdev.c]: Nice!

**Comment by bergzand** [comment]
Thanks for the review and testing!

**Pull request merged**